### PR TITLE
New Interfaces for Pinning Leadership

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch adds new interfaces that describe methods used for pinning leadership through lease management.

## QA steps

N/A

## Documentation changes

None yet, but the behaviour when implemented will need to be documented for commands that recruit it.

## Bug reference

N/A

### CODE REVIEW

File: core/leadership/interface.go [hunk 1]
```diff
@@ -38,6 +38,19 @@ type Claimer interface {
 	BlockUntilLeadershipReleased(applicationId string, cancel <-chan struct{}) (err error)
 }
 
+// Pinner describes methods used to manage suspension of application leadership
+// expiry.
+type Pinner interface {
+
+	// PinLeadership ensures that until "unpinned", the leadership of the input
+	// application will not change.
```
Comment (babbageclunk) [inline core/leadership/interface.go]: Won't expire, rather than won't change.

File: core/lease/interface.go [hunk 2]
```diff
@@ -46,6 +46,20 @@ type Claimer interface {
 	WaitUntilExpired(leaseName string, cancel <-chan struct{}) error
 }
 
+// Pinner describes methods used to manage suspension of lease expiry.
+type Pinner interface {
+
+	// Pin ensures that the current holder of input lease name will not loose
```
Comment (babbageclunk) [inline core/lease/interface.go]: Typo: loose should be lose.

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
$$merge$$ CI still in flux.

**Pull request merged**